lib/matplotlib/contour.py

```diff
@@ -405,8 +404,10 @@ def calc_label_rot_and_inline(self, slc, ind, lw, lc=None, spacing=5):
 
             ind = 0
 
-        # Path length in pixel space
-        pl = mlab.path_length(slc)
+        # Calculate path lengths
+        pl = np.zeros(slc.shape[0], dtype=float)
+        dx = np.diff(slc, axis=0)
+        pl[1:] = np.cumsum(np.hypot(dx[:, 0], dx[:, 1]))
```

Comment:
this expression is also going to allocate a temporary array (in cumsum) so you may as well do
```
np.row_stack([0, np.cumsum(...)])
```
(or concatenate, basically the same)
(in both cases there's two array allocations (three with dx))

Reply:
Can you apply
```
diff --git a/lib/matplotlib/contour.py b/lib/matplotlib/contour.py
index 9a14cf8e1..93d9516d7 100644
--- a/lib/matplotlib/contour.py
+++ b/lib/matplotlib/contour.py
@@ -407,7 +407,7 @@ class ContourLabeler(object):
         # Calculate path lengths
         pl = np.zeros(slc.shape[0], dtype=float)
         dx = np.diff(slc, axis=0)
-        pl[1:] = np.cumsum(np.hypot(dx[:, 0], dx[:, 1]))
+        pl = np.concatenate([[0], np.cumsum(np.hypot(*dx.T))])
         pl = pl - pl[ind]
 
         # Use linear interpolation to get points around label
diff --git a/lib/matplotlib/widgets.py b/lib/matplotlib/widgets.py
index 7100626ad..20755683c 100644
--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -1031,8 +1031,7 @@ class RadioButtons(AxesWidget):
 
         def inside(p):
             pcirc = np.array([p.center[0], p.center[1]])
-            d = pclicked - pcirc
-            return np.sqrt(np.dot(d, d)) < p.radius
+            return np.hypot(*(pclicked - pcirc)) < p.radius
 
         for i, (p, t) in enumerate(zip(self.circles, self.labels)):
             if t.get_window_extent().contains(event.x, event.y) or inside(p):
```
?

Reply:
Can you apply
```
diff --git a/lib/matplotlib/contour.py b/lib/matplotlib/contour.py
index 9a14cf8e1..274456fd4 100644
--- a/lib/matplotlib/contour.py
+++ b/lib/matplotlib/contour.py
@@ -405,9 +405,8 @@ class ContourLabeler(object):
             ind = 0
 
         # Calculate path lengths
-        pl = np.zeros(slc.shape[0], dtype=float)
         dx = np.diff(slc, axis=0)
-        pl[1:] = np.cumsum(np.hypot(dx[:, 0], dx[:, 1]))
+        pl = np.concatenate([[0], np.cumsum(np.hypot(*dx.T))])
         pl = pl - pl[ind]
 
         # Use linear interpolation to get points around label
diff --git a/lib/matplotlib/widgets.py b/lib/matplotlib/widgets.py
index 7100626ad..20755683c 100644
--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -1031,8 +1031,7 @@ class RadioButtons(AxesWidget):
 
         def inside(p):
             pcirc = np.array([p.center[0], p.center[1]])
-            d = pclicked - pcirc
-            return np.sqrt(np.dot(d, d)) < p.radius
+            return np.hypot(*(pclicked - pcirc)) < p.radius
 
         for i, (p, t) in enumerate(zip(self.circles, self.labels)):
             if t.get_window_extent().contains(event.x, event.y) or inside(p):
```
?